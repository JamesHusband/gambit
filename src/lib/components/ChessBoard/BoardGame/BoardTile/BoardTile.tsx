import { cn } from "../../../../utils";
import { ChessPiece } from "../ChessPiece";
import type { Piece } from "../../../../types";

export const BoardTile = ({
  rowIndex,
  colIndex,
  piece,
}: {
  position: string;
  rowIndex: number;
  colIndex: number;
  piece?: Piece | null;
}) => {
  const isDark = (rowIndex + colIndex) % 2 === 1;

  return (
    <div
      data-testid="board-tile"
      className={cn(
        "flex items-center justify-center w-full h-full",
        isDark ? "bg-[#769656]" : "bg-[#eeeed2]"
      )}
    >
      {piece && <ChessPiece piece={piece} />}
    </div>
  );
};
