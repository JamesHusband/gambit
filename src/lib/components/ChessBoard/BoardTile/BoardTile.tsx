import { cn } from "../../../utils";

export const BoardTile = ({
  rowIndex,
  colIndex,
}: {
  position: string;
  rowIndex: number;
  colIndex: number;
}) => {
  const isDark = (rowIndex + colIndex) % 2 === 1;

  return (
    <div
      data-testid="board-tile"
      className={cn(
        "flex items-center justify-center w-full h-full",
        isDark ? "bg-[#769656]" : "bg-[#eeeed2]"
      )}
    />
  );
};
