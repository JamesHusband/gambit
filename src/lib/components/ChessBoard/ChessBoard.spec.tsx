import { render } from "@testing-library/react";
import { ChessBoard } from ".";
import { BoardTile } from "./BoardTile";

jest.mock("../../utils", () => ({
  cn: (...args: string[]) => args.join(" "),
}));

jest.mock("./BoardTile", () => ({
  BoardTile: jest.fn(),
}));

jest.mock("../../hooks", () => ({
  useChessBoard: () => ({
    board: [[{ position: "a8" }], [{ position: "a7" }]],
  }),
}));

describe("ChessBoard", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render", () => {
    const { container } = render(<ChessBoard />);
    expect(container).toMatchSnapshot();
  });
  it("should render tiles", () => {
    render(<ChessBoard />);
    expect(BoardTile).toHaveBeenCalledTimes(2);
  });
});
