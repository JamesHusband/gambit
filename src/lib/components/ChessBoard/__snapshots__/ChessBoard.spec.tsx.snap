// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChessBoard should render 1`] = `
<div>
  <div
    class="flex justify-center items-center"
  >
    <div
      class="grid border border-gray-800 rounded-md w-full h-full"
      style="grid-template-rows: 32px 1fr 32px; grid-template-columns: 32px 1fr 32px;"
    >
      <div>
        BoardLabels
      </div>
      <div>
        BoardGame
      </div>
    </div>
  </div>
</div>
`;
